cmd/kubeadm/app/cmd/init.go

```diff
@@ -457,18 +459,105 @@ func (i *Init) Run(out io.Writer) error {
 	}
 
 	ctx := map[string]string{
-		"KubeConfigPath": adminKubeConfigPath,
 		"Token":          i.cfg.Token,
 		"CAPubKeyPin":    pubkeypin.Hash(caCert),
 		"MasterHostPort": masterHostPort,
 	}
+
+	kubeConfigInfo := fmt.Sprintf(`
+To start using your cluster, you need to run the following as a regular user:
+
+	  mkdir -p $HOME/.kube
+	  sudo cp -i %s $HOME/.kube/config
+	  sudo chown $(id -u):$(id -g) $HOME/.kube/config
+
+`, adminKubeConfigPath)
+
 	if i.skipTokenPrint {
 		ctx["Token"] = "<value withheld>"
 	}
 
+	ctx["KubeConfigInfo"] = kubeConfigInfo
+	if i.defaultUser != "" {
+		if err := copyConfigForDefaultUser(i.defaultUser, adminKubeConfigPath); err == nil {
+			ctx["KubeConfigInfo"] = ""
+		} else {
+			fmt.Printf("[init] WARNING: Could not copy the configuration for user %q: %v.\n", i.defaultUser, err)
+		}
+	}
+
 	return initDoneTempl.Execute(out, ctx)
 }
 
+func copyConfigForDefaultUser(defaultUser string, adminKubeConfigPath string) error {
+
+	var err error
+	var uid, gid int
+	var usr *user.User
+	var src, dest *os.File
+	var kubeDir, configFilePath string
+
+	fmt.Printf("[init] Copying configuration to %q from %q.\n", configFilePath, adminKubeConfigPath)
```

Comment:
configFilePath will always be `""` here, right?

Reply:
it might as well be. i will fix this.